docs/developer-docs/build/using-an-agent.mdx

```diff
@@ -0,0 +1,56 @@
+
+## Calling a Canister
+
+In the Internet Computer ecosystem, a library that is used to make calls to the IC public interface is called an agent. An agent has a few key responsibilities, which make it convenient to work with in your language of choice. 
+
+A list of agents is available at [other languages -> agents](./languages/other-languages/other-languages-intro#interfacing-with-canisters-agents).
+
+### 1. Structuring Data
+
+A `call` to the Internet Computer can take two common forms - an `update` or a `query`. The `agent` submits a POST request to `/api/v2/canister/<effective_canister_id>/call`, and includes the following components: 
+* `request_type`
+* Authentication
+    * `sender`, `nonce`, and `ingress_expiry`
+* `canister_id`
+* `method_name`
+* `arg` - the rest of the payload
+
+By knowing the Candid interface of the canister, the `agent` will assemble the `"arg"` by `cbor`-encoding the data from the client application, and ensuring it matches the Candid interface for the method it will be calling. That encoded data will be turned into a buffer.
+
+The `agent` takes all of that information, bundles it up into a certificate, and attaches it to the body of the POST request. The canister will work on that request asynchronously, and then the `agent` can begin polling with `read_state` requests, until the canister response is ready.
+
+### 2. Decoding Data
+
+Once the data has been returned from the IC, the `agent` takes the certificate from the payload and verifies it. The certificate can be verified as genuine using the public `rootKey` of the NNS subnet. The network will respond with a `cbor`-encoded buffer, which the `agent` can then decode, and transform into a useful structure using semantic language-specific types. For example, if the type returned from the canister is `text`, that will get turned into a JavaScript `string`, and so on.
+
+### 3. Managing Authentication
+
+Calls to the Internet Computer always need to have a cryptographic identity attached. That identity will either be Anonymous or Authenticated, using a cryptographic signature. Since identities are required, canisters can use the identity attached to a call to decide how to respond to that call. This enables contracts to use those identities for other purposes. 
+
+#### Accepted Identities
+
+The IC accepts calls using the following types of identities:
+
+* Ed25519
+* Secp256k1
+* ECDSA with a P-256 curve
+
+When encoding these identities as a Principal, agents attach a suffix byte, indicating whether the identity is self-authenticating or anonymous. 
+
+A self-authenticating identity using one of the above curves will have a suffix of 2, while anonymous identities receive a suffix of 4. The default anonymous identity is derived from an empty byte array, apart from the suffix 4. It resolves to `"2vxsx-fae"`, in its textual encoding.
```

Comment:
```suggestion
A self-authenticating identity using one of the above curves will have a suffix of 2.

While the anonymous identity is a single byte 4. It resolves to `"2vxsx-fae"`, in its textual encoding.
```

Only `0x04` is anonymous identity. Other bytes ending in 4 is not categorized so far.